sdk/data_kusto/src/authorization_policy.rs

```diff
@@ -0,0 +1,55 @@
+use azure_core::{auth::TokenCredential, Context, Policy, PolicyResult, Request};
+use http::header::AUTHORIZATION;
+use http::HeaderValue;
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct AuthorizationPolicy {
+    credential: Arc<dyn TokenCredential>,
+    resource: String,
+}
+
+impl std::fmt::Debug for AuthorizationPolicy {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        f.debug_struct("AuthorizationPolicy")
+            .field("credential", &"TokenCredential")
+            .field("resource", &self.resource)
+            .finish()
+    }
+}
+
+impl AuthorizationPolicy {
+    pub(crate) fn new<T>(credential: Arc<dyn TokenCredential>, resource: T) -> Self
+    where
+        T: Into<String>,
+    {
+        Self {
+            credential,
+            resource: resource.into(),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Policy for AuthorizationPolicy {
+    async fn send(
+        &self,
+        ctx: &Context,
+        request: &mut Request,
+        next: &[Arc<dyn Policy>],
+    ) -> PolicyResult {
+        assert!(
```

Comment:
Why assert over an error?

Reply:
This was copied from other pipelines. My understanding would be that this should either never or always happen at runtime, since creating the pipeline is crate internal. Thus - i think - the decision was made to just panic. @cataggar - is that correct?

Reply:
This is correct. Triggering this would mean a bug in the SDK code itself and thus we want to fail fast here.